randr/rroutput.c

```diff
@@ -466,86 +459,53 @@ ProcRRGetOutputInfo(ClientPtr client)
-            .type = X_Reply,
-            .status = RRSetConfigSuccess,
-            .sequenceNumber = client->sequence,
-            .length = bytes_to_int32(OutputInfoExtra),
-            .timestamp = pScrPriv->lastSetTime.milliseconds,
-            .crtc = output->crtc ? output->crtc->id : None,
-            .mmWidth = output->mmWidth,
-            .mmHeight = output->mmHeight,
-            .connection = output->nonDesktop ? RR_Disconnected : output->connection,
-            .subpixelOrder = output->subpixelOrder,
-            .nCrtcs = output->numCrtcs,
-            .nModes = output->numModes + output->numUserModes,
-            .nPreferred = output->numPreferred,
-            .nClones = output->numClones,
-            .nameLength = output->nameLength
-        };
-        extraLen = ((output->numCrtcs +
-                     output->numModes + output->numUserModes +
-                     output->numClones + bytes_to_int32(rep.nameLength)) << 2);
-
-        if (!extraLen)
-            goto sendout;
-
-        rep.length += bytes_to_int32(extraLen);
-        extra = calloc(1, extraLen);
-        if (!extra)
-            return BadAlloc;
-
-        crtcs = (RRCrtc *) extra;
-        modes = (RRMode *) (crtcs + output->numCrtcs);
-        clones = (RROutput *) (modes + output->numModes + output->numUserModes);
-
-        memcpy((clones + output->numClones), output->name, output->nameLength);
+        rep.crtc = output->crtc ? output->crtc->id : None;
+        rep.mmWidth = output->mmWidth;
+        rep.mmHeight = output->mmHeight;
+        rep.connection = output->nonDesktop ? RR_Disconnected : output->connection;
+        rep.subpixelOrder = output->subpixelOrder;
+        rep.nCrtcs = output->numCrtcs;
+        rep.nModes = output->numModes + output->numUserModes;
+        rep.nPreferred = output->numPreferred;
+        rep.nClones = output->numClones;
 
         for (i = 0; i < output->numCrtcs; i++) {
-            crtcs[i] = output->crtcs[i]->id;
-            if (client->swapped)
-                swapl(&crtcs[i]);
+            x_rpcbuf_write_CARD32(&rpcbuf, output->crtcs[i]->id);
         }
 
         for (i = 0; i < output->numModes + output->numUserModes; i++) {
             if (i < output->numModes)
-                modes[i] = output->modes[i]->mode.id;
+                x_rpcbuf_write_CARD32(&rpcbuf, output->modes[i]->mode.id);
             else
-                modes[i] = output->userModes[i - output->numModes]->mode.id;
-            if (client->swapped)
-                swapl(&modes[i]);
+                x_rpcbuf_write_CARD32(&rpcbuf, output->userModes[i - output->numModes]->mode.id);
         }
+
         for (i = 0; i < output->numClones; i++) {
-            clones[i] = output->clones[i]->id;
-            if (client->swapped)
-                swapl(&clones[i]);
+            x_rpcbuf_write_CARD32(&rpcbuf, output->clones[i]->id);
         }
+
+        x_rpcbuf_write_string_pad(&rpcbuf, output->name);
```

Comment:
Is it executed at the end of both branches of that big `if` above? We should probably move it outside of the `if` to avoid repetition.